features/dump_offline_inventories.lua

```diff
@@ -11,13 +11,30 @@ local Config = require 'config'
 local set_timeout_in_ticks = Task.set_timeout_in_ticks
 local config = Config.dump_offline_inventories
 
+local ignored_items_set = {}
+for _ , k in pairs(config.ignored_items) do
+    ignored_items_set[k] = true
+end
+
 local offline_player_queue = {}
 
 Global.register({offline_player_queue = offline_player_queue}, function(tbl)
     offline_player_queue = tbl.offline_player_queue
     config = Config.dump_offline_inventories
 end)
 
+local function move_items(source, target)
+    if not source.is_empty() then
+        for i = 1, #source do
+            local stack = source[i]
+            if stack.valid_for_read and not stack.is_selection_tool() and not stack.is_blueprint_book() and not ignored_items_set[stack.name] then
```

Comment:
```suggestion
            if stack.valid and stack.valid_for_read and not stack.is_selection_tool and not stack.is_blueprint_book and not ignored_items_set[stack.name] then
```
According to the API these are boolean values not functions.
https://lua-api.factorio.com/latest/LuaItemStack.html#LuaItemStack.is_selection_tool
https://lua-api.factorio.com/latest/LuaItemStack.html#LuaItemStack.is_blueprint_book

Also I think bizarrely you have to check valid as well as valid for read.

Reply:
Got carried over by the ``is_empty()``

As for ``stack`` being invalid. That would imply that the inventory we pull from contains invalid references which should never be an constitute a major bug in Factorio.

Reply:
There might be an edge case if the stack gets cleared on the same tick by another event. The stack might still be there but not valid? I'm not sure, but the Factorio devs always recommend adding the valid check and in this case there is no performance concern and it doesn't make the code more complicated. So, I'd be happier with the check, same above for the inventory check.